e3sm_supported_machines/create_new_cime_env.bash

```diff
@@ -21,10 +21,10 @@ check_env () {
 
 # Modify the following to choose which cime-env version(s)
 # the python version(s) are installed
-versions=(1.7.0)
-pythons=(3.9)
+versions=(1.8.0)
+pythons=(3.12)
 
-default_python=3.9
+default_python=3.12
```

Comment:
Seems like we might as well be using python 3.12.  @mkstratos, any reason you think this isn't a good idea?

Reply:
All the tests I have pass for evv using Python 3.12, so I think we should be okay.  Just to note though, one downstream dependency for LIVVkit/evv4esm (pybtex) had an issue with `pkg_resources` being removed, but adding `setuptools` as a dependency fixes this. (It's been fixed in their repository, but they haven't released a new version yet).

Reply:
That's good information.  I will back off to 3.11 if this causes trouble but I think it's best to find out...